server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/api/DeleteCondition.java

```diff
@@ -112,6 +109,24 @@ public final int hashCode() {
         }
     }
 
+    class WithEnqueueId implements DeleteCondition {
+        private final EnQueueId enQueueId;
+
+
+        public WithEnqueueId(EnQueueId enQueueId) {
+            this.enQueueId = enQueueId;
+        }
+
+        public EnQueueId getEnQueueId() {
+            return enQueueId;
+        }
+
+        @Override
+        public boolean shouldBeDeleted(Mail mail) {
+            throw new NotImplementedException("EnQueueId is not carried as a Mail property");
```

Comment:
so why did you bother to implement this condition?

Reply:
It's fixed later, I intended to 'keep the code changes light'